tests/components/modern_forms/test_config_flow.py

```diff
@@ -0,0 +1,198 @@
+    assert result2.get("type") == RESULT_TYPE_CREATE_ENTRY
+
+    assert "data" in result2
+    assert result2["data"][CONF_HOST] == "192.168.1.123"
+    assert result2["data"][CONF_MAC] == "AA:BB:CC:DD:EE:FF"
+
+
+@patch(
+    "homeassistant.components.modern_forms.ModernFormsDevice.update",
+    side_effect=ModernFormsConnectionError,
+)
+async def test_connection_error(
+    update_mock: MagicMock, hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test we show user form on Modern Forms connection error."""
+    aioclient_mock.post("http://example.com/mf", exc=aiohttp.ClientError)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_HOST: "example.com"},
+    )
+
+    assert result.get("type") == RESULT_TYPE_FORM
+    assert result.get("step_id") == "user"
+    assert result.get("errors") == {"base": "cannot_connect"}
+
+
+@patch(
+    "homeassistant.components.modern_forms.ModernFormsDevice.update",
+    side_effect=ModernFormsConnectionError,
+)
+async def test_zeroconf_connection_error(
+    update_mock: MagicMock, hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test we abort zeroconf flow on Modern Forms connection error."""
+    aioclient_mock.post("http://192.168.1.123/mf", exc=aiohttp.ClientError)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data={"host": "192.168.1.123", "hostname": "example.local.", "properties": {}},
+    )
+
+    assert result.get("type") == RESULT_TYPE_ABORT
+    assert result.get("reason") == "cannot_connect"
+
+
+@patch(
+    "homeassistant.components.modern_forms.ModernFormsDevice.update",
+    side_effect=ModernFormsConnectionError,
+)
+async def test_zeroconf_confirm_connection_error(
+    update_mock: MagicMock, hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test we abort zeroconf flow on Modern Forms connection error."""
+    aioclient_mock.post("http://192.168.1.123:80/mf", exc=aiohttp.ClientError)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": SOURCE_ZEROCONF,
+            CONF_HOST: "example.com",
+            CONF_NAME: "test",
+        },
+        data={"host": "192.168.1.123", "hostname": "example.com.", "properties": {}},
+    )
+
+    assert result.get("type") == RESULT_TYPE_ABORT
+    assert result.get("reason") == "cannot_connect"
+
+
+async def test_user_device_exists_abort(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test we abort zeroconf flow if Modern Forms device already configured."""
+    aioclient_mock.post(
+        "http://192.168.1.123:80/mf",
+        text=load_fixture("modern_forms/device_info.json"),
+        headers={"Content-Type": CONTENT_TYPE_JSON},
+    )
+
+    await init_integration(hass, aioclient_mock)
```

Comment:
We don't need to set up the whole integration here. Adding the mock config entry is enough. Pass the skip parameter.

Reply:
```py
await init_integration(hass, aioclient_mock, skip_setup=True)
```